telemetry/specs/INT.mdk

```diff
@@ -377,6 +377,44 @@ Considering all these factors, the INT specification requires that an INT
 switch **must not fragment** packets in order to append INT information to
 the packet.
 
+### Checksum Update
+
+INT may be transported over an L4 protocol such as TCP or UDP, or over an
+encapsulation header that includes an L4 header, such as VXLAN. In these cases,
+an INT source or transit switch may be required to update the L4 Checksum of
+the encapsulating header.
+
+In some cases, an L4 Checksum update is not necessary. For example, when UDP is 
+transported over IPv4, it is possible to assign a zero Checksum, causing the 
+receiver to ignore the value of the Checksum field. For UDP over IPv6, there
+are specific use cases in which it is possible to assign a zero Checksum 
+(as defined in RFC 6936).
+
+If an L4 Checksum update is required, an INT source/transit switch may perform
+it in one of two ways:
+
+* Update the L4 Checksum field such that the new value is equal to the Checksum 
+of the new packet, after the INT-related updates, or
+
+* Assign a value to the Checksum Complement metadata which guarantees that the 
+existing L4 Checksum is the correct value of the packet after the INT-related 
+updates.
+
+The motivation for the Checksum Complement is that some hardware implementations
+process data packets in a serial order, which may impose a problem when INT
+fields and metadata that reside after the L4 Checksum field are inserted or 
+modified. Therefore, the Checksum Complement metadata, if present, is the last
+metadata field in the stack. 
+
```

Comment:
We should add some clarification here that the INT source indicates that checksum neutral updates are allowed by setting an instruction bit corresponding to checksum complement metadata. The above text on INT transit being able to perform L4 checksum update in one of two ways seems to indicate that this is always the case. If the INT source does not set the checksum complement instruction bit, then transit hops have to update the L4 checksum (except if L4 checksum is zero for VXLAN or GENEVE in which case checksum update can be skipped altogether).